compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -637,13 +638,9 @@ static LogicalResult setRootConfig(
   if (isAArch64(*variantOp) && !isQuantized) {
     return setAArch64RootConfig(entryPointFn, contractionOp, flowTileSizes,
                                 workgroupTileSizes, vectorSize);
-  } else if (isX86(*variantOp) || isRISCV(*variantOp) ||
-             isAArch64(*variantOp)) {
-    if (disableMatmulPadPipeline || numLoops != 3) {
-      return setMatmulNoPadRootConfig(entryPointFn, contractionOp,
-                                      flowTileSizes, workgroupTileSizes,
-                                      vectorSize);
-    }
+  } else if (disableMatmulPadPipeline) {
```

Comment:
So this enables padding for RISC-V for example... Is that desirable. We have no way of tracking what impact that has on RISC-V. Padding is pretty heavy weight, i'd keep that mostly for x86 for now.

Reply:
For RISC-V side, I kept the padding approach for only x86 in the beginning, but @dcaballe had a point to keep x86 and RISC-V aligned. I'm not sure if we still want to make them align today, I'll let @dcaballe make the call.

For unknown platform side, do we prefer using non-padding approach? Maybe it's better because we won't put heavy weight to unknown stuff.

Reply:
Yeah I'd say keep the padding approach default for just the x86 path.

Reply:
Sorry, I left my comments yesterday but I forgot to click on submit! 

If we disable padding on RISC-V we would be regressing on all the GEMMs that we have claimed to be vectorizing already. I'm not sure I get why RISC-V should be different than x86 in this sense. Happy to give it a quick try to these changes on RISC-V if you tell me a model where padding can make an large impact or you send me the binaries. If we are not sure yet about the impact/side-effects of padding, perhaps we should reconsider enabling it by default? We could keep it in a way that we can use it for search and make the decision later. If we finally have to enable this transformation for only one target, we could do it in a way that the pipelines do not diverge but it's more about setting a flag (my comment https://github.com/google/iree/pull/1 in https://github.com/google/iree/pull/9310#issuecomment-1158372565 goes toward that direction). Otherwise, it will get difficult to introduce new transformations to all the CPU targets. WDYT?